plugins/wasi_nn/neuralspeed.cpp

```diff
@@ -0,0 +1,338 @@
+#include <dlfcn.h>
+#endif
+namespace WasmEdge::Host::WASINN::NeuralSpeed {
+#ifdef WASMEDGE_PLUGIN_WASI_NN_BACKEND_NEURAL_SPEED
+#if defined(_WIN32) || defined(_WIN64) || defined(__WIN32__) ||                \
+    defined(__TOS_WIN__) || defined(__WINDOWS__)
+HINSTANCE SharedLib = LoadLibrary(PYTHON_LIB_PATH);
+#else
+void *SharedLib = dlopen(PYTHON_LIB_PATH, RTLD_GLOBAL | RTLD_NOW);
+#endif
+void printImformation(Graph &GraphRef, Context &CxtRef) {
+  spdlog::info(
+      "[WASI-NN][Info] Neural speed backend: Number of input tokens: {}"sv,
+      CxtRef.Inputs.size());
+  spdlog::info(
+      "[WASI-NN][Info] Neural speed backend: Number of Output tokens: {}"sv,
+      CxtRef.Outputs.size());
+  spdlog::info("[WASI-NN][Info] Neural speed backend: Load time: {} ms"sv,
+               GraphRef.LoadTime);
+  spdlog::info("[WASI-NN][Info] Neural speed backend: Compute time: {} ms "sv,
+               GraphRef.ComputeTime);
+}
+Expect<WASINN::ErrNo> load(WASINN::WasiNNEnvironment &Env,
+                           Span<const Span<uint8_t>> Builders, WASINN::Device,
+                           uint32_t &) noexcept {
+  // Add a new graph.
+  Env.NNGraph.emplace_back(Backend::NeuralSpeed);
+  auto &GraphRef = Env.NNGraph.back().get<Graph>();
+  const auto StartTime = std::chrono::steady_clock::now();
+  // Initialize the plugin parameters.
+  GraphRef.EnableDebugLog = true;
+  if (GraphRef.EnableDebugLog) {
+    spdlog::info("[WASI-NN][Debug] Neural speed backend: Load."sv);
+  }
+
+  if (Builders.size() > 1) {
+    std::string Metadata = std::string(
+        reinterpret_cast<char *>(Builders[1].data()), Builders[1].size());
+    simdjson::dom::parser Parser;
+    simdjson::dom::element Doc;
+    auto ParseError = Parser.parse(Metadata).get(Doc);
+    if (ParseError) {
+      spdlog::error("[WASI-NN] Neural speed backend: Parse metadata error"sv);
+      return ErrNo::InvalidEncoding;
+    }
+    if (Doc.at_key("model_type").error() == simdjson::SUCCESS) {
+      std::string_view model_type;
+      auto Err = Doc["model_type"].get<std::string_view>().get(model_type);
+      if (Err) {
+        spdlog::error(
+            "[WASI-NN] Neural speed backend: Unable to retrieve the model_type option."sv);
+        return ErrNo::InvalidArgument;
+      }
+      GraphRef.model_type = model_type;
+    }
+  }
+
+  // Handle the model path.
+  auto Weight = Builders[0];
+  const std::string BinModel(reinterpret_cast<char *>(Weight.data()),
+                             Weight.size());
+  spdlog::info("[WASI-NN][Debug] Neural speed: BinModel: {}"sv,
+               BinModel.size());
+  std::string ModelFilePath;
+  if (BinModel.substr(0, 8) == "preload:") {
+    ModelFilePath = BinModel.substr(8);
+  } else {
+    if (GraphRef.EnableDebugLog) {
+      spdlog::info(
+          "[WASI-NN][Debug] Neural speed: Model path not found in nn-preload, "
+          "write model into a tmpfile."sv);
+    }
+    // TODO: pass the model directly to ggml
```

Comment:
I would like to know the details of this TODO item.

Reply:
It refers to [ggml backend](https://github.com/WasmEdge/WasmEdge/blob/30c7553a773f8e495a856201ea4c37973bad6622/plugins/wasi_nn/ggml.cpp#L744). Although current implementation is enough. Should I remove it?

Reply:
I have changed. Please check, thank you.